.github/workflows/main.yml

```diff
@@ -7,30 +7,46 @@ on:
     # run daily, this refreshes the cache
     - cron: '13 2 * * *'
 
+# Cancel a currently running workflow from the same PR, branch or tag when
+# a new workflow is triggered:
+# https://stackoverflow.com/questions/66335225/how-to-cancel-previous-runs-in-the-pr-when-you-push-new-commitsupdate-the-curre
+concurrency:
```

Comment:
Please add this change into its own separate commit, it's unrelated to the python changes